### DISCUSSION TITLE
v3.3.0  v4.0.2 api update tips.

### DISCUSSION THREAD

**Discussion by xlinx** [theoephraim/node-google-spreadsheet#635] [General]
v3.3.0  v4.0.2

Auth part
when add JWT auth to api will get 401 error,
add new scopes: drive
```
    let serviceAccountAuth = new JWT({
      email: process.env.GOOGLE_SERVICE_ACCOUNT_EMAIL,
      key: process.env.GOOGLE_PRIVATE_KEY.replace(/\\n/g, "\n"),
      scopes: [
        'https://www.googleapis.com/auth/spreadsheets',
        'https://www.googleapis.com/auth/drive'   //<------- example not list this, add this work all great.
      ],
    });
```
working with row and get column value diff

```
let r = await sheetTemp.getRows();
//google sheet api v3.3.0
let cellVal=r[index][columnName];
//google sheet api v4.0.2
let cellVal=r[index].get(columnName);
```

for the people update from v3 to v4...

**Comment by theoephraim** [comment]
Have you seen the upgrade guide? https://theoephraim.github.io/node-google-spreadsheet/#/guides/migration-guide

In many cases, you will only need drive scope if you are doing operations like managing permissions or creating / deleting docs.
